### ISSUE TITLE
Leaking ServiceConnection

### ISSUE THREAD

**Issue by BigPun86** [mauron85/react-native-background-geolocation#2]
Hey there, i get this error message a lot of time. And i am not sure how to handle this or how to fix this....

````
E  Activity com.bal.bla.MainActivity has leaked ServiceConnection com.marianhello.react.BackgroundGeolocationModule$1@39337a07 that was originally bound here
E  android.app.ServiceConnectionLeaked: Activity com.bla.bla.MainActivity has leaked ServiceConnection com.marianhello.react.BackgroundGeolocationModule$1@39337a07 that was originally bound here
E      at android.app.LoadedApk$ServiceDispatcher.<init>(LoadedApk.java:1077)
E      at android.app.LoadedApk.getServiceDispatcher(LoadedApk.java:971)
E      at android.app.ContextImpl.bindServiceCommon(ContextImpl.java:1782)
E      at android.app.ContextImpl.bindService(ContextImpl.java:1765)
E      at android.content.ContextWrapper.bindService(ContextWrapper.java:539)
E      at com.marianhello.react.BackgroundGeolocationModule.doBindService(BackgroundGeolocationModule.java:277)
E      at com.marianhello.react.BackgroundGeolocationModule.startAndBindBackgroundService(BackgroundGeolocationModule.java:242)
E      at com.marianhello.react.BackgroundGeolocationModule.start(BackgroundGeolocationModule.java:215)
E      at java.lang.reflect.Method.invoke(Native Method)
E      at java.lang.reflect.Method.invoke(Method.java:372)
E      at com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke(BaseJavaModule.java:319)
E      at com.facebook.react.bridge.NativeModuleRegistry$ModuleDefinition.call(NativeModuleRegistry.java:187)
E      at com.facebook.react.bridge.NativeModuleRegistry.call(NativeModuleRegistry.java:62)
E      at com.facebook.react.bridge.CatalystInstanceImpl$NativeModulesReactCallback.call(CatalystInstanceImpl.java:432)
E      at com.facebook.react.bridge.queue.NativeRunnableDeprecated.run(Native Method)
E      at android.os.Handler.handleCallback(Handler.java:739)
E      at android.os.Handler.dispatchMessage(Handler.java:95)
E      at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
E      at android.os.Looper.loop(Looper.java:135)
E      at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:193)
E      at java.lang.Thread.run(Thread.java:818)

````

Any ideas on this?

RN 29.2
Android

**Comment by mauron85** [comment]
Can you try latest version?

**Comment by BigPun86** [comment]
@mauron85 Sorry for the late reply. I am currently using rn 29.2 and the modules version is at 0.1.1. I will try the alpha today

**Comment by mauron85** [comment]
I've identified the cause, but I'm worried it can not be fixed until react-native issue [#9791] (https://github.com/facebook/react-native/issues/9791) is fixed. Should not be show stopper. It only occurs when destroying/terminating app (in principle you can ignore it).

**Comment by mauron85** [comment]
should be fixed in react-native 0.33. just upgrade your react-native to 0.33
